syntax = "proto3";

package Db;

// user_data
message UserBase
{
   string user_name = 1;
   int64 update_time = 2;
}

// user_data key=(count_type,count_subtype)
message UserCount
{
   int32 count_type = 1;
   int32 count_subtype = 2;
   int32 count_value = 3;
   int64 count_tick = 4;
}

// user_data key=cache_type
message ClientCache
{
   int64 cache_type = 1;
   int64 modify_time = 2;
   map<string, string> cache_items = 3;
}

message UserData
{
   string user_id = 1;
   UserBase user_base = 2;
   repeated UserCount user_counts = 3;
   repeated ClientCache client_cache = 4;

}

message KVData
{
   UserData user_data = 1;
}